event/src/unvalidated.rs

```diff
@@ -0,0 +1,249 @@
+    pub fn cap(&self) -> Option<Cid> {
+        // Parse signatures[0].protected for cad
+        self.signatures[0].protected.as_ref().and_then(|protected| {
+            serde_json::from_slice::<Protected>(protected.as_slice())
+                .ok()
+                .and_then(|protected| protected.cap)
+        })
+    }
+}
+
+/// Payload of a signed event
+#[derive(Debug, Serialize, Deserialize)]
+// Note untagged variants a deserialized in order and the first one that succeeds is returned.
+// Therefore the order of the variants is important to be most specific to least specific
+#[serde(untagged)]
+pub enum Payload<D> {
+    /// Data event
+    Data(DataPayload<D>),
+    /// Init event
+    Init(InitPayload<D>),
+}
+
+/// Payload of an init event
+#[derive(Debug, Serialize, Deserialize)]
+pub struct InitPayload<D> {
+    header: InitHeader,
+    data: Option<D>,
+}
+
+impl<D> InitPayload<D> {
+    /// Construct a new payload for an init event
+    pub fn new(header: InitHeader, data: Option<D>) -> Self {
+        Self { header, data }
+    }
+
+    /// Get the header
+    pub fn header(&self) -> &InitHeader {
+        &self.header
+    }
+
+    /// Get the data
+    pub fn data(&self) -> Option<&D> {
+        self.data.as_ref()
+    }
+}
+
+/// Headers for an init event
+#[derive(Debug, Serialize, Deserialize)]
+pub struct InitHeader {
+    controllers: Vec<String>,
+    sep: String,
+    model: Bytes,
+    unique: Option<Bytes>,
+}
+
+impl InitHeader {
+    /// Construct a header for an init event payload
+    pub fn new(controllers: Vec<String>, sep: String, model: Bytes, unique: Option<Bytes>) -> Self {
+        Self {
+            controllers,
+            sep,
+            model,
+            unique,
+        }
+    }
+
+    /// Get the controllers
+    pub fn controllers(&self) -> &[String] {
+        self.controllers.as_ref()
+    }
+
+    /// Get the sep
+    pub fn sep(&self) -> &str {
+        self.sep.as_ref()
+    }
+
+    /// Get the model
+    pub fn model(&self) -> &Bytes {
+        &self.model
+    }
+}
+
+/// Payload of a data event
+#[derive(Debug, Serialize, Deserialize)]
+pub struct DataPayload<D> {
+    id: Cid,
+    prev: Cid,
+    header: Option<DataHeader>,
+    data: D,
+}
+
+impl<D> DataPayload<D> {
+    /// Construct a new payload for a data event
+    pub fn new(id: Cid, prev: Cid, header: Option<DataHeader>, data: D) -> Self {
+        Self {
+            id,
+            prev,
+            header,
+            data,
+        }
+    }
+
+    /// Get the id
+    pub fn id(&self) -> Cid {
+        self.id
+    }
+    /// Get the prev
+    pub fn prev(&self) -> &Cid {
+        &self.prev
+    }
+    /// Get the header
+    pub fn header(&self) -> Option<&DataHeader> {
+        self.header.as_ref()
+    }
+    /// Get the data
+    pub fn data(&self) -> &D {
+        &self.data
+    }
+}
+
+/// Headers for a data event
+#[derive(Debug, Serialize, Deserialize)]
+pub struct DataHeader {
```

Comment:
`shouldIndex` can show up in a DataHeader.  Do we need to specify that field explicitly if it can show up, even if rust-ceramic never needs to inspect it?

Reply:
Yes, what data type is it? A boolean?

Reply:
yep, boolean